src/Namshi/JOSE/JWT.php

```diff
@@ -34,9 +34,9 @@ class JWT
      */
     public function __construct(array $payload, array $header)
     {
-        $this->payload = $payload;
-        $this->header  = $header;
-        $this->encoder = new Base64UrlSafeEncoder();
+        $this->setPayload($payload);
```

Comment:
I was wrong, this is to make sure that the method of the subclass is called which does have special logic. For instance in EasyJWS `setPayload` adds the 'iat' claim.